### DISCUSSION TITLE
[Feat]: Make package available for svelte@next 5.0.0

### DISCUSSION THREAD

**Discussion by LiddellHunter** [davidjerleke/embla-carousel#796] [Ideas] [answered]
### Which variants of Embla Carousel are relevant to this feature request?

- [ ] embla-carousel **(Core)**
- [ ] embla-carousel-react
- [ ] embla-carousel-vue
- [X] embla-carousel-svelte
- [ ] embla-carousel-autoplay
- [ ] embla-carousel-auto-scroll
- [ ] embla-carousel-solid
- [ ] embla-carousel-auto-height
- [ ] embla-carousel-class-names
- [ ] embla-carousel-docs **(Documentation)**
- [ ] embla-carousel-docs **(Generator)**

### Feature description

I'm working in Svelte 5 and would love to have this available in that. If it's as simple as changing the package svelte dep to "|| ^5.0.0"  I would greatly appreciate it!. 

### Additional Context

Additional details here...

### Before submitting

- [X] I've made research efforts and searched the documentation
- [X] I've searched for existing issues
- [x] I agree to follow this project's Contributing Guidelines for feature requests

**Comment by davidjerleke** [comment]
Hi @LiddellHunter,

Thank you for your feature request. I haven’t read the release notes for Svelte v5. But feel free to contribute here and read the release notes. The only Svelte specific thing that `embla-carousel-svelte` is using is the structure of an [ActionReturn](https://github.com/davidjerleke/embla-carousel/blob/bcd5f643fd07ed7c78c156abb906aada0449f29d/packages/embla-carousel-svelte/src/components/emblaCarouselSvelte.ts#L1). If v5 has breaking changes for `ActionReturn`, we need to update the code. If not, I guess it’s just a matter of doing what you suggest:

> If it's as simple as changing the package svelte dep to "|| ^5.0.0" I would greatly appreciate it!.

Additionally, I think we need to test it with v5 and simply see if it’s working as expected. 

Best,
David

**Reply by davidjerleke** [reply]
@LiddellHunter do you want to pursue this? Did you [read my comment](https://github.com/davidjerleke/embla-carousel/discussions/796#discussioncomment-8809818)?

**Comment by davidjerleke** [comment]
Anyone stumbling across this, feel free to make a contribution and please read [this comment](https://github.com/davidjerleke/embla-carousel/discussions/796#discussioncomment-8809818) before.

**Reply by shyakadavis** [reply]
Hello, @davidjerleke 

Hunter, in upgrading shadcn-svelte to Svelte 5, noted that everything is fine as is. Have a look at this comment here: https://github.com/huntabyte/shadcn-svelte/pull/1182

And I guess he didn't run into any issues, otherwise he would have reported, right?

**Reply by davidjerleke** [reply]
@shyakadavis thanks for helping out! Let's ask him directly and see if he is available and has any input. @huntabyte did you run into any problems using Embla Carousel with Svelte 5? Please see [this comment for context](https://github.com/davidjerleke/embla-carousel/discussions/796#discussioncomment-8809818).

**Reply by huntabyte** [reply]
I did not have any issues functionality-wise whatsoever! Perhaps the peer dependency message but that was expected.

**Reply by davidjerleke** [reply]
@huntabyte thanks for confirming. Svelte 5 support has been released in v8.3.0 so if you upgrade your package the dependency warning should go away. See:

- https://github.com/davidjerleke/embla-carousel/pull/992

**Comment by S1r-Lanzelot** [comment]
https://github.com/davidjerleke/embla-carousel/pull/992 will look to address this in the coming week or so but please see the use of overrides while 5.0 is still in `next` phase.

**Comment by davidjerleke** [comment]
Anyone who wants to use bleeding edge Svelte will have to use this setting in their project:

```json
"overrides": {
  "embla-carousel-svelte": {
    "svelte": "^5.0.0-next.1"
  }
}
```

Having said that, the next Embla release includes Svelte 5 support thanks to @S1r-Lanzelot:

- #992 

Closing this as resolved.

**Reply by davidjerleke** [reply, marked as answer]
Svelte 5 support was just released in [v8.3.0](https://github.com/davidjerleke/embla-carousel/releases/tag/v8.3.0).
